### PR TITLE
RGW Zipper - clean up and clarify lifecycle of s->object, s->bucket, and friends

### PR DESCRIPTION
**Pull request by dang**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/workunits/rgw/test_librgw_file.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ then
     export AWS_ACCESS_KEY_ID=`openssl rand -base64 20`
     export AWS_SECRET_ACCESS_KEY=`openssl rand -base64 40`
 
-    radosgw-admin user create --uid ceph-test-librgw-file \
+    radosgw-admin ${CEPH_CONF} user create --uid ceph-test-librgw-file \
```
Comment (cbodley) [inline qa/workunits/rgw/test_librgw_file.sh:9]: ceph already supports the `CEPH_CONF` environment variable (see `getenv("CEPH_CONF")` in src/common/config.cc)

i would expect this script to work without changes, if you invoke it with `CEPH_CONF=/path/to/ceph.conf`

Comment (dang) [reply qa/workunits/rgw/test_librgw_file.sh:9]: Interesting.  I believe it didn't, but I'll try again, and if it does, remove this commit.

Comment (dang) [reply qa/workunits/rgw/test_librgw_file.sh:9]: I must have done something wrong, it works fine.

**Review by cbodley — APPROVED** [review]

File: qa/workunits/rgw/test_librgw_file.sh (hunk 1 above)
Comment (cbodley) [reply qa/workunits/rgw/test_librgw_file.sh]: \o/

**Review by soumyakoduri — APPROVED** [review]

**Comment by dang** [comment]
https://pulpito.ceph.com/dang-2021-09-17_12:24:18-rgw-wip-dang-zipper-lifecycle-distro-basic-smithi/

**Pull request merged**